src/lib.rs

```diff
@@ -68,12 +68,16 @@ impl FixedBitSet
         }
     }
     /// Create a new **FixedBitSet** with a specific number of bits,
-    /// all initially clear.
-    pub fn with_capacity_and_blocks(bits: usize, data: Vec<Block>) -> Self
+    /// initialized from provided blocks.
+    ///
+    /// This will panic if the blocks are not the exact size needed
+    /// for the capacity.
+    pub fn with_capacity_and_blocks<I: IntoIterator<Item=Block>>(bits: usize, blocks: I) -> Self
     {
-        let (mut blocks, rem) = div_rem(bits, BITS);
-        blocks += (rem > 0) as usize;
-        assert_eq!(blocks, data.len());
+        let (mut n_blocks, rem) = div_rem(bits, BITS);
+        n_blocks += (rem > 0) as usize;
+        let data: Vec<Block> = blocks.into_iter().collect();
```

Comment:
Thanks. Here's our consolation - Vec::into_iter().collect() roundtrips without extra allocation/copying (in this case where the iterator is fresh, no elements taken).